lib/travis/build/script/elm.rb

```diff
@@ -0,0 +1,183 @@
+            ' in the issue', ansi: :green
+
+
+          sh.if '! -d sysconfcpus/bin' do
+            install_sysconfcpus
+          end
+
+          sh.fold 'install.elm' do
+            install_elm
+            install_elm_test
+            install_elm_format
+          end
+        end
+
+        def script
+          sh.cmd 'elm-format --validate . && elm-test'
+        end
+
+        def setup_cache
+          if data.cache?(:elm)
+            sh.fold 'cache.elm' do
+              # Cache the ~/.elm directory.
+              directory_cache.add '$HOME/.cache/elm', '$HOME/.elm'
+
+              directory_cache.add '$HOME/.cache/elm-stuff', 'elm-stuff'
+
+              if elm_major_version == 0 && elm_minor_version <= 18
+                # In Elm 0.18, some put their tests in ./test instead of ./tests
+                directory_cache.add '$HOME/.cache/elm-test-stuff', 'test/elm-stuff'
+              end
+
+              # In Elm 0.18+, all tests live in tests/ by default
+              # (whereas previously tests/ was allowed, but so was test/)
+              directory_cache.add '$HOME/.cache/elm-tests-stuff', 'tests/elm-stuff'
+
+              # we build sysconfcpus from source, so cache the result
+              directory_cache.add '$HOME/.cache/sysconfcpus-cache', 'sysconfcpus'
+            end
+          end
+        end
+
+        def cache_slug
+          super << '-elm-' << elm_version
+        end
+
+        private
+
+          def elm_version
+            config[:elm].to_s
+          end
+
+          def elm_version_number(index)
+            elm_version.split(".")[index]
+          end
+
+          def elm_major_version
+            elm_version_number 0
+          end
+
+          def elm_minor_version
+            elm_version_number 1
+          end
+
+          def elm_patch_version
+            elm_version_number 2
+          end
+
+          def elm_test_version
+            config[:elm_test].to_s
+          end
+
+          def elm_format_version
+            config[:elm_format].to_s
+          end
+
+          def install_elm
+            npm_install "-g elm@elm-#{elm_version}"
+
+            convert_binary_to_sysconfcpus 'elm'
+
+            # Beginning with Elm 0.19, there's one `elm` binary and that's it.
+            # In that case, there won't be any files to convert here!
+            if elm_major_version == 0 && elm_minor_version <= 18
+              convert_binary_to_sysconfcpus 'elm-make'
+              convert_binary_to_sysconfcpus 'elm-package'
+              convert_binary_to_sysconfcpus 'elm-format'
+            end
+          end
+
+          def install_elm_format
+            npm_install "-g elm-format@elm-#{elm_format_version}"
+
+            convert_binary_to_sysconfcpus 'elm-format'
+          end
+
+          def install_elm_test
+              sh.if "$(vers2int $(echo `node --version` | tr -d 'v')) -lt $(vers2int #{ELM_TEST_REQUIRED_NODE_VERSION})" do
```

Comment:
`vers2int` has been renamed to `travis_vers2int` (to explicit the origin of the function).

Reply:
Updated in 294455217d26c8a01fe8aa979fa5e5a8127ead9c